source-stripe-native/source_stripe_native/api.py

```diff
@@ -0,0 +1,495 @@
+from datetime import datetime, timedelta, UTC
+from decimal import Decimal
+from estuary_cdk.http import HTTPSession
+from logging import Logger
+from typing import Iterable, Any, Callable, Awaitable, AsyncGenerator
+import json
+
+from estuary_cdk.capture.common import (
+    PageCursor,
+    LogCursor,
+)
+
+
+from .models import EventResult, BackfillResult, ListResult
+
+
+API = "https://api.stripe.com/v1"
+
+async def fetch_incremental(
+    cls,
+    http: HTTPSession,
+    log: Logger,
+    log_cursor: LogCursor,
+) -> AsyncGenerator:
+    """ Note: Stripe's data is always served in reverse-chronological order.
+    fetch_incremental works by accessing stripe's Events API
+    Each Resource that contains a valid Event Type is passed here.
+    It works by calling the Events API, parsing the Events result
+    and validating the incoming model class.
+    If the document was created after the last log_cursor, 
+    yield this document and later yield the newest log_cursor.
+    """
+    stop = True
+
+    url = f"{API}/events"
+    parameters = {"type": cls.TYPES, "limit": 100}
+    recent = []
+
+
+    _cls: Any = cls  # Silence mypy false-positive
+
+    while stop:
```

Comment:
It's pretty confusing that while a variable called `stop` is `true`, the loop continues, and only when `stop` is set to `false` does the loop stop.

Reply:
Makes a lot of sense, changed this variable name to `iterating` in 00467212d8135d9c433b5806262295c55c368f4c